src/propagation/constant_integer_propagator.cpp

```diff
@@ -0,0 +1,181 @@
+#include "propagation_visitor.h"
+#include "constant_propagator.h"
+
+#include <clangmetatool/propagation/constant_integer_propagator.h>
+
+#include <string>
+
+#include <clang/Analysis/CFG.h>
+#include <clang/AST/Expr.h>
+#include <llvm/ADT/APSInt.h>
+
+namespace clangmetatool {
+namespace propagation {
+namespace {
+
+// Given a qualified type, determine if it is an int
+inline bool isIntegerType(const clang::QualType& QT) {
```

Comment:
This can be made to a `isNumericType()` to cover all numeric types into this PR. Other changes would be simple renames, as i understand it. Would appreciate suggestions.

Reply:
F2F: Better to keep these implementations separate